cmd/cpe2cve/config.go

```diff
@@ -0,0 +1,300 @@
+	// profiling
+	cpuProfile string
+	memProfile string
+	// feeds
+	overrides multiString
+
+	feeds map[string][]string // provider -> [file]
+}
+
+func (c *config) addFlags() {
+	// input
+	flag.IntVar(&c.cpesAt, "cpe", 0, "look for CPE names in input at this position (starts with 1)")
+
+	// output
+	flag.IntVar(&c.cvesAt, "cve", 0, "output CVEs at this position (starts with 1)")
+	flag.IntVar(&c.matchesAt, "matches", 0, "output CPEs that matches CVE at this position; 0 disables the output")
+	flag.IntVar(&c.cwesAt, "cwe", 0, "output problem types (CWEs) at this position (starts with 1)")
+	flag.IntVar(&c.cvss2at, "cvss2", 0, "output CVSS 2.0 base score at this position (starts with 1)")
+	flag.IntVar(&c.cvss3at, "cvss3", 0, "output CVSS 3.0 base score at this position (starts with 1)")
+	flag.IntVar(&c.cvssAt, "cvss", 0, "output CVSS base score (v3 if available, v2 otherwise) at this position (starts with 1)")
+	flag.Var(&c.skip, "e", "comma separated list of fields to erase from output; starts at 1, supports ranges (e.g. 1-3); processed before the vulnerablitie field added")
+	flag.StringVar(&c.provider, "provider", "", "feed provider. affects the output only when the provider_field is specified: adds the $provider into that field")
+	flag.IntVar(&c.providerAt, "provider_field", 0, "where should the provider be placed in the output (starts with 1).")
+
+	// separators
+	flag.StringVar(&c.inFieldSep, "d", "\t", "input columns delimiter")
+	flag.StringVar(&c.inRecSep, "d2", ",", "inner input columns delimiter: separates elements of list passed into a CSV columns")
+	flag.StringVar(&c.outFieldSep, "o", "\t", "output columns delimiter")
+	flag.StringVar(&c.outRecSep, "o2", ",", "inner output columns delimiter: separates elements of lists in output CSV columns")
+
+	// optimizations
+	flag.IntVar(&c.nProcessors, "nproc", 1, "number of concurrent goroutines that perform CVE lookup")
+	flag.BoolVar(&c.indexedDict, "idxd", false, "build and use an index for CVE dictionary: increases the processing speed, but might miss some matches")
+	flag.Int64Var(&c.cacheSize, "cache_size", 0, "limit the cache size to this amount in bytes; 0 removes the limit, -1 disables caching")
+	flag.BoolVar(&c.requireVersion, "require_version", false, "ignore matches of CPEs with version ANY")
+
+	// profiling
+	flag.StringVar(&c.cpuProfile, "cpuprofile", "", "file to store CPU profile data to; empty value disables CPU profiling")
+	flag.StringVar(&c.memProfile, "memprofile", "", "file to store memory profile data to; empty value disables memory profiling")
+
+	// feeds
+	flag.Var(&c.overrides, "r", "overRide: path to override feed, can be specified multiple times")
+}
+
+func (c *config) addFeedsFromArgs(args ...string) error {
+	if c.feeds == nil {
+		c.feeds = make(map[string][]string)
+	}
+	if len(args) > 0 && c.providerAt != 0 && c.provider == "" {
+		return fmt.Errorf("in order to output a provider, you need to specify a valid provider name using the -provider flag")
```

Comment:
feels like this check belongs to `validate()` below

Reply:
Moved